src/main/java/appeng/client/gui/implementations/GuiPatternMulti.java

```diff
@@ -0,0 +1,252 @@
+        this.buttonList.add(
+                this.symbolSwitch = new GuiImgButton(
+                        this.guiLeft + 22,
+                        this.guiTop + 53,
+                        Settings.ACTIONS,
+                        ActionItems.MULTIPLY));
+
+        ItemStack myIcon = null;
+        final Object target = ((AEBaseContainer) this.inventorySlots).getTarget();
+        final IDefinitions definitions = AEApi.instance().definitions();
+        final IParts parts = definitions.parts();
+
+        if (target instanceof PartPatternTerminal) {
+            for (final ItemStack stack : parts.patternTerminal().maybeStack(1).asSet()) {
+                myIcon = stack;
+            }
+            this.originalGui = GuiBridge.GUI_PATTERN_TERMINAL;
+        }
+
+        if (target instanceof PartPatternTerminalEx) {
+            for (final ItemStack stack : parts.patternTerminalEx().maybeStack(1).asSet()) {
+                myIcon = stack;
+            }
+            this.originalGui = GuiBridge.GUI_PATTERN_TERMINAL_EX;
+        }
+
+        if (this.originalGui != null && myIcon != null) {
+            this.buttonList.add(
+                    this.originalGuiBtn = new GuiTabButton(
+                            this.guiLeft + 154,
+                            this.guiTop,
+                            myIcon,
+                            myIcon.getDisplayName(),
+                            itemRender));
+        }
+
+        this.amountToSet = new GuiTextField(
+                this.fontRendererObj,
+                this.guiLeft + 50,
+                this.guiTop + 57,
+                59,
+                this.fontRendererObj.FONT_HEIGHT);
+        this.amountToSet.setEnableBackgroundDrawing(false);
+        this.amountToSet.setMaxStringLength(16);
+        this.amountToSet.setTextColor(GuiColors.CraftAmountToCraft.getColor());
+        this.amountToSet.setVisible(true);
+        this.amountToSet.setFocused(true);
+        this.amountToSet.setText(String.valueOf(0));
+        this.amountToSet.setSelectionPos(0);
+    }
+
+    @Override
+    public void drawFG(final int offsetX, final int offsetY, final int mouseX, final int mouseY) {
+        this.fontRendererObj
+                .drawString(GuiText.SelectAmount.getLocal(), 8, 6, GuiColors.CraftAmountSelectAmount.getColor());
+    }
+
+    @Override
+    public void drawBG(final int offsetX, final int offsetY, final int mouseX, final int mouseY) {
+        this.set.displayString = GuiText.Set.getLocal();
+
+        this.bindTexture("guis/patternMulti.png");
+        this.drawTexturedModalRect(offsetX, offsetY, 0, 0, this.xSize, this.ySize);
+
+        try {
+            String out = this.amountToSet.getText();
+
+            double resultD = Calculator.conversion(out);
+            int resultI;
+
+            if (Double.isNaN(resultD)) {
+                resultI = 0;
```

Comment:
Why does default value sometimes equal 1 and sometimes - 0? Should it equal 1 everywhere (to prevent zero multiplication and division, cause both don't make sense), shouldn't it?

Reply:
It's an oversight on my part, but I think 0 is a better default than 1, because 0 means nothing will be done, but 1 will be multiplied by 1.